pkg/snapshot/snapshotter/snapshotter.go

```diff
@@ -290,7 +290,9 @@ func (ssr *Snapshotter) takeFullSnapshot() (*brtypes.Snapshot, error) {
 	if ssr.prevSnapshot.Kind == brtypes.SnapshotKindFull && ssr.prevSnapshot.LastRevision == lastRevision {
 		ssr.logger.Infof("There are no updates since last snapshot, skipping full snapshot.")
 	} else {
-		ctx, cancel = context.WithTimeout(context.TODO(), ssr.etcdConnectionConfig.ConnectionTimeout.Duration)
+		// Note: As FullSnapshot can be very large, so to avoid context timeout use "LongerConnectionTimeout" in context.WithTimeout()
+		// refer: https://github.com/gardener/etcd-backup-restore/issues/360
+		ctx, cancel = context.WithTimeout(context.TODO(), ssr.etcdConnectionConfig.LongerConnectionTimeout.Duration)
```

Comment:
```suggestion
		// Note: As FullSnapshot can be very large, so to avoid context timeout use "SnapshotTimeout" in context.WithTimeout()
		ctx, cancel = context.WithTimeout(context.TODO(), ssr.etcdConnectionConfig.SnapshotTimeout.Duration)
```